File: app/src/main/java/org/netdex/androidusbscript/service/LuaUsbServiceConnection.java
package org.netdex.androidusbscript.service;

import android.content.ComponentName;
import android.content.ServiceConnection;
import android.os.IBinder;

public class LuaUsbServiceConnection implements ServiceConnection {
    private LuaUsbService service_;

    public LuaUsbServiceConnection() {
    }

    @Override
    public void onServiceConnected(ComponentName name, IBinder binder) {
        service_ = ((LuaUsbService.Binder) binder).getService();
    }

    @Override
    public void onServiceDisconnected(ComponentName name) {
        service_ = null;
    }

    public LuaUsbService getService(){
        return service_;
    }
}

File: app/src/main/java/org/netdex/androidusbscript/service/LuaUsbServiceConnection.kt
package org.netdex.androidusbscript.service

import android.content.ComponentName
import android.content.ServiceConnection
import android.os.IBinder

open class LuaUsbServiceConnection : ServiceConnection {
    var service: LuaUsbService? = null
        private set

    override fun onServiceConnected(name: ComponentName, binder: IBinder) {
        service = (binder as LuaUsbService.Binder).service
    }

    override fun onServiceDisconnected(name: ComponentName) {
        service = null
    }
}
